### ISSUE TITLE
Mjolnir bot polling when it doesn't possess Admin

### ISSUE THREAD

**Issue by pterocles** [spantaleev/matrix-docker-ansible-deploy#2147]
**Describe the bug**
In 2f48740525419424ea8a49e89bbee4885878cbd7 a change was introduced that causes Mjolnir to poll endlessly into oblivion.

**To Reproduce**
My `vars.yml` file looks like this:

```yaml
---
# The bare domain name which represents your Matrix identity.
# Matrix user ids for your server will be of the form (`@user:<matrix-domain>`).
#
# Note: this playbook does not touch the server referenced here.
# Installation happens on another server ("matrix.<matrix-domain>").
#
# If you've deployed using the wrong domain, you'll have to run the Uninstalling step,
# because you can't change the Domain after deployment.
#
# Example value: example.com
matrix_domain: <secret>
matrix_server_fqn_matrix: <secret>

# matrix_synapse_admin_enabled: true
# matrix_synapse_admin_container_http_host_bind_port: '127.0.0.1:6666'
# matrix_synapse_container_client_api_host_bind_port: '127.0.0.1:6667'

# The Matrix homeserver software to install.
# See `roles/matrix-base/defaults/main.yml` for valid options.
matrix_homeserver_implementation: synapse

# A secret used as a base, for generating various other secrets.
# You can put any string here, but generating a strong one is preferred (e.g. `pwgen -s 64 1`).
matrix_homeserver_generic_secret_key: '<secret>'

# This is something which is provided to Let's Encrypt when retrieving SSL certificates for domains.
#
# In case SSL renewal fails at some point, you'll also get an email notification there.
#
# If you decide to use another method for managing SSL certificates (different than the default Let's Encrypt),
# you won't be required to define this variable (see `docs/configuring-playbook-ssl-certificates.md`).
#
# Example value: someone@example.com
matrix_ssl_lets_encrypt_support_email: '<secret>'

# A Postgres password to use for the superuser Postgres user (called `matrix` by default).
#
# The playbook creates additional Postgres users and databases (one for each enabled service)
# using this superuser account.
matrix_postgres_connection_password: '<secret>'
matrix_client_element_enabled: false

# Optimizations for small VPS
# An identity server is not a must.
matrix_ma1sd_enabled: false

# Disabling this will prevent email-notifications and other such things from working.
matrix_mailer_enabled: false

# You can also disable this to save more RAM,
# at the expense of audio/video calls being unreliable.
matrix_coturn_enabled: false

# Amount of time to check if everything is okay when starting services
matrix_common_after_systemd_service_start_wait_for_timeout_seconds: 60

# Nginx optimizations
matrix_nginx_proxy_proxy_matrix_client_api_forwarded_location_synapse_admin_api_enabled: false
matrix_nginx_proxy_proxy_matrix_nginx_status_enabled: false
matrix_nginx_proxy_access_log_enabled: false
# This makes Synapse not keep track of who is online/offline.
#
# Keeping track of this and announcing such online-status in federated rooms with
# hundreds of servers inside is insanely heavy (https://github.com/matrix-org/synapse/issues/3971).
#
# If your server does not federate with hundreds of others, enabling this doesn't hurt much.
matrix_synapse_presence_enabled: false
matrix_synapse_log_level: "CRITICAL"
matrix_synapse_storage_sql_log_level: "CRITICAL"
matrix_synapse_root_log_level: "CRITICAL"
matrix_synapse_caches_global_factor: 2.0
matrix_synapse_configuration_extension_yaml: |
  matrix_synapse_retention:
    enabled: true
    default_policy:
      min_lifetime: 1d
      max_lifetime: 7d
    allowed_lifetime_min: 1d
    allowed_lifetime_max: 7d
    purge_jobs:
     - longest_max_lifetime: 1d
       interval: 1d
     - shortest_max_lifetime: 7d
       interval: 1d


  # Resource-constrained homeserver settings
  #
  # When this is enabled, the room "complexity" will be checked before a user
  # joins a new remote room. If it is above the complexity limit, the server will
  # disallow joining, or will instantly leave.
  #
  # Room complexity is an arbitrary measure based on factors such as the number of
  # users in the room.
  #
  limit_remote_rooms:
  # Uncomment to enable room complexity checking.
  #
    enabled: true

  # the limit above which rooms cannot be joined. The default is 1.0.
  #
    complexity: 10

  # override the error which is returned when the room is too complex.
  #
    complexity_error: "This room is too complex."

  # allow server admins to join complex rooms. Default is false.
  #
  # admins_can_join: true

#  postgres optimizations

matrix_postgres_process_extra_arguments: [
  "-c shared_buffers=512MB",
  "-c effective_cache_size=1536MB",
  "-c effective_io_concurrency=200",
  "-c random_page_cost=1.1",
  "-c min_wal_size=1GB",
  "-c maintenance_work_mem=128MB",
  "-c checkpoint_completion_target=0.9",
  "-c wal_buffers=16MB",
  "-c default_statistics_target=100",
  "-c work_mem=1310kB",
  "-c max_wal_size=4GB",
  ]

# mjolnir

matrix_bot_mjolnir_enabled: true
### This token is not an Admin token becasue Mjolnir doesn't require Admin on this HS
matrix_bot_mjolnir_access_token: "<token>"
matrix_bot_mjolnir_management_room: "!ID:matrix.org"
matrix_bot_mjolnir_configuration_extension_yaml: |
  # Your custom YAML configuration goes here.
  # This configuration extends the default starting configuration (`matrix_bot_mjolnir_configuration_yaml`).
  #
  # You can override individual variables from the default configuration, or introduce new ones.
  #
  # If you need something more special, you can take full control by
  # completely redefining `matrix_bot_mjolnir_configuration_yaml`.

  # homeserverUrl: http://matrix-synapse:8008
  # rawHomeServerUrl: http://matrix-synapse:8008

  # Misc options for command handling and commands
  commands:
  # If true, Mjolnir will respond to commands like !help and !ban instead of
  # requiring a prefix. This is useful if Mjolnir is the only bot running in
  # your management room.
  #
  # Note that Mjolnir can be pinged by display name instead of having to use
  # the !mjolnir prefix. For example, "my_moderator_bot: ban @spammer:example.org"
  # will ban a user.
    allowNoPrefix: true
  #
  # automatic redact on certain ban keywords
  automaticallyRedactForReasons:
    - "spam"
    - "scam"
    - "gore"
    - "ban evasion"
    - "illicit"
    - "hate speech"
    - "illegal"

  # show invite requests in management room
  recordIgnoredInvites: true

  # Options for exposing web APIs.
  # web:
    # Whether to enable web APIs.
    # enabled: true

    # The port to expose the webserver on. Defaults to 8080.
    # port: 8080

    # The address to listen for requests on. Defaults to only the current
    # computer.
    # address: localhost

    # Alternative setting to open to the entire web. Be careful,
    # as this will increase your security perimeter:
    #
    #  address: "0.0.0.0"

    # A web API designed to intercept Matrix API
    # POST /_matrix/client/r0/rooms/{roomId}/report/{eventId}
    # and display readable abuse reports in the moderation room.
    #
    # If you wish to take advantage of this feature, you will need
    # to configure a reverse proxy, see e.g. test/nginx.conf
    # abuseReporting:
    # Whether to enable this feature.
    #   enabled: true

    # Whether or not to actively poll synapse for abuse reports, to be used
    # instead of intercepting client calls to synapse's abuse endpoint, when that
    # isn't possible/practical.
    # pollReports: true

    # Whether or not new reports, received either by webapi or polling,
    # should be printed to our managementRoom.
    # displayReports: true
```

**Expected behavior**
Update Mjolnir config seeing as there was a change to the role.

**Matrix Server:**
 - OS: Debian 11, Bullseye
 - Architecture: amd64

**Additional context**
The breaking change was accidentally introduced in #2141:

Probably assumed each Mjolnir bot would posses Admin privs.

```bash
commit 2f48740525419424ea8a49e89bbee4885878cbd7
Merge: 66ef8614 59c46662
Author: Slavi Pantaleev <slavi@devture.com>
Date:   Sat Oct 1 14:21:14 2022 +0300

    Merge pull request #2141 from FSG-Cat/Mjolnir-CFG-Update

    Update Mjolnir Config from its #962 state to todays template.
```

**Comment by spantaleev** [comment]
Should be fixed in #2146

**Comment by pterocles** [comment]
It is!
